jfrog-cli/utils/cliutils/utils.go

```diff
@@ -114,9 +118,51 @@ func confirmAnswer(answer string) bool {
 }
 
 func GetVersion() string {
+	if ok, _ := utils.GetBoolEnvValue("JFROG_CLI_SHOW_UPDATE", true); ok {
+		latestRelease, err := getLatestRelease()
+		if err != nil {
+			return CliVersion
+		}
+
+		if CliVersion == latestRelease {
+			return fmt.Sprintf("%s (you're running the latest version)", CliVersion)
+		}
+		return fmt.Sprintf("%s (there is a newer version available [v%s])", CliVersion, latestRelease)
+	}
 	return CliVersion
 }
 
+func getLatestRelease() (string, error) {
```

Comment:
1. It is probably best to get the latest JFrog CLI version from Bintray,
2. We prefer to use JFrog CLI's http client infrastructure, rather than duplicating the functionality. Since we'll be using Bintray to get the latest version, then we can add a new GetLatestVersion API to the jfrog-client-go library used by jfrog-cli-go. This new API can be then used by anyone, similarly to all the other APIs documented in the [jfrog-client-go's README](https://github.com/jfrog/jfrog-client-go/blob/master/README.md).

Reply:
Great feedback, absolutely agree with what you mentioned. I do have a question on the BinTray API for that. If I read the documentation of the API correctly, I'd need to call the endpoint for `GetVersion`. That would be equivalent to calling

```go
versionPath, err := versions.CreatePath("jfrog/jfrog-cli-go/<package>/_latest")
btManager.ShowVersion(versionPath)
```

According to the API docs, it does require an authenticated user and I'm not sure which user that should be as not everyone will have access to that repository. Thoughts are welcome!

Reply:
The existing *jfrog-client-go* Bintray API may require authentication, but the Bintray's *get version* API does not.
You can see ti by running
```
curl -XGET https://api.bintray.com/packages/jfrog/jfrog-cli-go/jfrog-cli-linux-amd64/versions/_latest
```
which returns
```
{"name":"1.23.1","desc":null,"package":"jfrog-cli-linux-amd64","repo":"jfrog-cli-go","owner":"jfrog","labels":[],"published":true,"attribute_names":[],"created":"2018-12-26T04:39:03.073Z","updated":"2018-12-26T04:45:17.197Z","released":"","ordinal":51.0,"rating_count":0}
```
So we may need to improve the http client infrastructure to allow non authenticated requests to Bintray.
Let me know if you need any help with this 😄